### PR TITLE
dev-cmd/bump: don't use Repology version if livecheckable

### PR DESCRIPTION
**Pull request by dawidd6**
- [ ] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

This should fix issues like one seen in https://github.com/Homebrew/homebrew-core/pull/116956 when the Repology version is greater than the one we get from Livecheck and the strategy is different than GithubLatest.

Basically if we define a livecheck block, I think we want the Livecheck to be the top priority over Repology.
If we don't define the livecheck block, then Repology data should be equal to what Livecheck returns.

### CODE REVIEW

**Review by MikeMcQuaid — APPROVED** [review]
Makes sense to me, thanks @dawidd6!

**Review by chenrui333 — APPROVED** [review]
